src/display/annotation_layer.js

```diff
@@ -198,6 +199,92 @@ class AnnotationElement {
     return AnnotationElement._hasPopupData(this.data);
   }
 
+  get hasCommentButton() {
+    return this.enableComment && this._isEditable && this.hasPopupElement;
+  }
+
+  get commentButtonPosition() {
+    const { quadPoints, rect } = this.data;
+    let maxX = -Infinity;
+    let maxY = -Infinity;
+    if (quadPoints?.length >= 4) {
```

Comment:
Shouldn't this be 8 instead of 4 since each quadrilateral has 8 coordinates, and we also skip that many indexes in the loop?